src/hydra/resourceSchemaAnalyzer.js

```diff
@@ -0,0 +1,68 @@
+const getReferenceNameField = reference => {
```

Comment:
This entire file is hard to read and would greatly benefit from input/output examples in jsDoc.